engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -4786,6 +4798,12 @@ struct shear_t : public demon_hunter_attack_t
         p()->proc.soul_fragment_from_meta->occur();
       }
     }
+
+    if ( p()->buff.t30_vengeance_4pc->up() )
+    {
+      p()->active.fiery_brand_t30->execute_on_target( s->target );
+      p()->buff.t30_vengeance_4pc->reset();
```

Comment:
->reset() is for end of iteration and resets some other things like last_start, last_trigger, etc. Should just use ->expire() here.